import "building.proto";
import "resource.proto";
import "hero.proto";

package SanProto;

message ConscriptInfo {
  required int32 building_basic_id = 1;     //征兵建筑
  required int32 city_basic_id = 2;         //建筑所在的城池
  required int32 location_index = 3;        //建筑在城池中的位置
  required int32 current_soldier_num = 4;   //兵营当前自由的存兵数量
  required int32 lock_soldier_num = 5;      //兵营当前锁定的兵数量
  required int32 conscript_num = 6;         //此次征兵过程的征兵数
  optional int32 total_conscript_time = 7;  //此次征兵过程的总时间
  optional int32 conscript_passed_time = 8; //此次征兵过程经过的时间
}

message StartConscriptReq {
  required uint64 user_id = 1;
  required ConscriptInfo conscript = 2;       //征兵信息
  optional BuildingInfo building = 3;         //征兵建筑
  repeated HeroInfo heroes = 4;               //变更的英雄
  repeated BuildingInfo buildings = 5;        //变更的建筑
}

message StartConscriptRes {
  required int32 status = 1;
  optional ResourceInfo resource = 2;
}

message EndConscriptReq {
  required uint64 user_id = 1;
  required ConscriptInfo conscript = 2; //征兵信息
  required BuildingInfo building = 3;   //征兵建筑
  repeated HeroInfo heroes = 4;         //英雄结算
  optional bool use_gold = 5;         //玩家是否使用元宝立即完成
}

message EndConscriptRes {
  enum RET {
    OK = 0;
    NOT_CONSCRIPTING = 1;
    CANNT_END = 2;
  }
  required int32 status = 1;
  optional ResourceInfo resource = 2;
  optional RET ret = 3;
  optional ConscriptInfo conscript = 4;
  optional BuildingInfo building = 5;
}

